django/__init__.py

```diff
@@ -10,9 +13,15 @@ def setup(set_prefix=True):
     Configure the settings (this happens as a side effect of accessing the
     first setting), configure logging and populate the app registry.
     Set the thread-local urlresolvers script prefix if `set_prefix` is True.
+
+    .. versionchanged:: 5.0
+
+        Raise a ``RuntimeWarning`` when apps interact with the database until
+        the app registry has been fully populated.
```

Comment:
As far as I know, we don't tend to add `versionchanged` notes in docstrings. These are best placed in the documentation (as you already did). Perhaps just remove the `versionchanged` line and leave the `Raises ...`?